src/spikeinterface/core/base.py

```diff
@@ -298,6 +298,23 @@ def to_dict(
         dump_dict: dict
             Serialized dictionary
         """
+        to_dict_kwargs = dict(
+            include_annotations=include_annotations,
+            include_properties=include_properties,
+            relative_to=relative_to,
+            folder_metadata=folder_metadata,
+        )
+
+        new_kwargs = dict()
+        transform_extractors_to_dict = lambda x: x.to_dict(**to_dict_kwargs) if isinstance(x, BaseExtractor) else x
+        for name, value in self._kwargs.items():
+            if isinstance(value, list):
+                new_kwargs[name] = [transform_extractors_to_dict(element) for element in value]
+            if isinstance(value, dict):
```

Comment:
This second if is wrong. It should be an elif. But if I correct this, the performance degrades to the one of https://github.com/SpikeInterface/spikeinterface/pull/1622.